### PR TITLE
small regression: [com_menus] items view: use empty for all menu instead of *

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
#### Summary of Changes

The "All Menu Items" was implemented in https://github.com/joomla/joomla-cms/pull/10190 (which used empty as the value for the "- Select Menu -" option), but after the menus ACL was implemented in https://github.com/joomla/joomla-cms/pull/9814 (which changed to `*` the value for the "- Select Menu -" option).

So, this PR goes back to (empty) as the value for the "- Select Menu -" option.

This allows to use `/administrator/index.php?option=com_menus&view=items&menutype=`URL (without a 403) as a fallback (in case the menu type is not found or something). This URL lists the all menu items list view.
#### Testing Instructions
1. Use latest staging
2. Go to `/administrator/index.php?option=com_menus&view=items&menutype=` you will get a 403 error
3. Now apply patch
4. Go to `/administrator/index.php?option=com_menus&view=items&menutype=` you'll be redirect to "All Menu Items" with no 403 error
5. Navigate through the menus and create/Edit menu items and check if you're properly redirected after without issues
6. Test batch and other operations also to confirm all is ok.

@infograf768 @bembelimen since you're the original creators of the two PR mentioned please test if all work as it should.
#### Observations

After this PR if you use:
- `/administrator/index.php?option=com_menus&view=items` you'll be redirect to your current state selected menu item ("All Menu Items" if none).
- `/administrator/index.php?option=com_menus&view=items&menutype=` you'll be redirect to "All Menu Items"
- `/administrator/index.php?option=com_menus&view=items&menutype=validmenutype` you'll be redirect to your valid menu type
- `/administrator/index.php?option=com_menus&view=items&menutype=invalidmenutype` you'll get a error the menu is not found
- `/administrator/index.php?option=com_menus&view=items&menutype=validmenutypebutnopermission` you'll get a error not autorized

### CODE REVIEW

**Comment by Webdongle** [comment]
@andrepereiradasilva 
I am confused about the test instructions.

Before applying the patch all test from 2 - 8 work as expected.

If you remove steps 1 and 2 from the test instructions
Then after the Test instructions you have
**Expected result**
describe what should be seen

**Actual result**
describe the error

Then .. After applying patch describe what error is fixed.

It would be easier to test.  Because at the moment I am unable to reproduce the error this is supposed to fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by andrepereiradasilva** [comment]
ok sorry, test instructions updated, now i think it's easier

**Comment by Webdongle** [comment]
I have tested this item :white_check_mark: successfully on ac541f63f1a87e28243481ba40c7bd6df4f1dcf5<br /><br />Patch works

FULL SUCCESS<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on ac541f63f1a87e28243481ba40c7bd6df4f1dcf5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by infograf768** [comment]
RTC. Has to go into 3.6.0 as this is a new bug introduced by https://github.com/joomla/joomla-cms/pull/9814<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by bembelimen** [comment]
Hello,
```/administrator/index.php?option=com_menus&view=items&menutype=existing-menu-without-access-rights```
Expected result: 403
Current result: show all items

**Comment by bembelimen** [comment]
I have tested this item :red_circle: unsuccessfully on ac541f63f1a87e28243481ba40c7bd6df4f1dcf5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by brianteeman** [comment]
Oops didnt see the failed test -set back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by Webdongle** [comment]
@bembelimen 
>/administrator/index.php?option=com_menus&view=items&menutype=existing-menu-without-access-rights

What menu "menu-without-access-rights" ?  If a user has access to the Menus >>> Manage >>> Menus view then they have access to all menus.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by brianteeman** [comment]
There is a new acl feature in 3.6

**Comment by andrepereiradasilva** [comment]
@bembelimen thanks for checking that. Will check.

**Comment by infograf768** [comment]
i tested tha acl and it was working fine. here.

**Comment by andrepereiradasilva** [comment]
@bembelimen please check now.
The code changes were this: https://github.com/joomla/joomla-cms/pull/10459/commits/a86f1844a0dc0fa690b53b6079419801047ba10e

**Comment by andrepereiradasilva** [comment]
BTW @bembelimen there are some questions i notice in the menus ACL:

When you have a user that can't do anything (all denied) for a particular menu type:
 - that user can edit that menu type (bug?)
 - he can click the counting icons but not the all items (menu type title) (bug?)
 - that menu doesn't appear in the admin menu list, nor the select menu selector.
 - also that menu items doesn't appear in the all menu items (with the edit options disabled).

In other words the menus ACL doesn't behave like the categories ACL (or other ACL) and IMHO it should. That i know of, normally a user without permissions can view the items are there, but can't change them.

**Comment by andrepereiradasilva** [comment]
Update: Sorry i had "Access Administration Interface" set to Denied.

But still that is one issue (when all set to denied). that user can edit that menu type (bug?)

**Comment by BurtNL** [comment]
I have tested this item :white_check_mark: successfully on a86f1844a0dc0fa690b53b6079419801047ba10e<br /><br />I have applied this patch on the latest staging together with #10460.
It is working as described. So this test is successfull.

I have not tested the new ACL feature.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by bembelimen** [comment]
Hello @andrepereiradasilva 

> But still that is one issue (when all set to denied). that user can view/edit that menu type in Menu Types Manage (bug?)

The access rights are for the item(s) view only, not for the menu type stuff itself, there work the regular ACL settings

**Comment by andrepereiradasilva** [comment]
> The access rights are for the item(s) view only, not for the menu type stuff itself, there work the regular ACL settings

@bembelimen 
I see it was implemented that way, but IMO doesn't make much sense a user can delete the entire menu and don't have access to delete a menu item in the same menu.

I made some changes to implement, at least part, of the ACL also in the menu types list view.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on a86f1844a0dc0fa690b53b6079419801047ba10e<br /><br />Except that we do not get a 403 but a 500 with message "You are not authorised to view this resource", the changes do now work when the menutype has no access rights when using: 
>/administrator/index.php?option=com_menus&view=items&menutype=existing-menu-without-access-rights

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by infograf768** [comment]
RTC

Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by infograf768** [comment]
Conflicts concern only one file.

**Comment by andrepereiradasilva** [comment]
conflicts fixed

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 8e294e3bb8dbe7b9d5bbf4401f912027170acfb9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by roland-d** [comment]
Taking RTC off as we only have one successful test so far<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by bembelimen** [comment]
I have tested this item :white_check_mark: successfully on 8e294e3bb8dbe7b9d5bbf4401f912027170acfb9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@roland-d RTC now?

thanks for testing @infograf768 @bembelimen

**Comment by brianteeman** [comment]
Back to rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10459">issues.joomla.org/joomla-cms/10459</a>.</sub>

**Comment by roland-d** [comment]
I am looking into it

**Comment by roland-d** [comment]
Thanks everybody
